### ISSUE TITLE
kaniko push takes time

### ISSUE THREAD

**Issue by vinodhkumart** [GoogleContainerTools/kaniko#2822]
Hi , 
I am running a jenkins pipeline where kaniko bake just takes 4 mins of time and kaniko push takes 30 mins of time while pushing it to Quay... However, i tried to run the same set of push commands manually inside the container and it just takes only 4 mins to push the image to quay.

Did anyone faced such issues earlier? please let me know..

Below is the output where it just wait in for 30mins and push the image:

2:22:43  [36mINFO[0m[0180] ENTRYPOINT [ "npm", "run", "serve_static"]   
12:22:43  [37mDEBU[0m[0180] mapping stage idx 0 to digest sha256:e1e2634c64fdf45hy7893e20909d6dbch78gdffd2bcfaaf4a30f1605515578b7ff4 
12:22:43  [37mDEBU[0m[0180] mapping digest sha256:e1e2634c64fdf45hy7893e20909d6dbch78gdffd2bcfaaf4a30f1605515578b7ff4  to cachekey

**Comment by vinodh1506** [comment]
Is any one looking into this?

**Comment by bstrdsmkr** [comment]
@JeromeJu are you by chance using the --reproducible flag? If so, that does a lot of work just before the push step and can look like the push is taking forever when it's actually the reproducible work that's going on

**Comment by vinodh1506** [comment]
@bstrdsmkr No. That flag is being used for kaniko baking only which takes 4 mins of time but not for push.
**Kaniko baking:**
/kaniko/executor '--dockerfile=Dockerfile' --context /home/jenkins/workspace/release_testing --destination quay.io/mytesting/test:1.0 --cleanup --single-snapshot --no-push --reproducible '--tarPath=test.tar'

**kaniko publish:**
/kaniko/executor '--dockerfile=Dockerfile' --context /home/jenkins/workspace/release_testing --destination quay.io/mytesting/test:1.0 --image-name-with-digest-file image-name-with-digest-file
